src/Adapter/Cache/CacheClearer.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * 2007-2017 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Cache;
+
+use Tools;
+use Media;
+
+/**
+ * Class able to clear application caches
+ */
+class CacheClearer
+{
+    /**
+     * Clear all application caches
+     */
+    public function clearAllCaches()
+    {
+        Tools::clearSf2Cache();
+        Tools::clearSmartyCache();
+        Tools::clearXMLCache();
+        Media::clearCache();
+        Tools::generateIndex();
```

Comment:
With this call we are not cleaning the cache.

Don't you think we should have a CacheWarmer somewhere?

Reply:
Hmm at this point, this is not expected to clear the Symfony cache here. We should probably use CacheWarmer, not in the adapter but in the corresponding "new" class (this need a refactoring of how PrestaShop manages to clear his cache(s), it's a really big topic :)  )